v2/datastream-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/DataStreamToSpanner.java

```diff
@@ -567,26 +608,71 @@ public static PipelineResult run(Options options) {
               .apply("Reshuffle", Reshuffle.viaRandomKey());
     }
     /*
-     * Stage 2: Write records to Cloud Spanner
+     * Stage 2: Transform records
      */
+
     // Ingest transformation context file into memory.
     TransformationContext transformationContext =
         TransformationContextReader.getTransformationContext(
             options.getTransformationContextFilePath());
-    SpannerTransactionWriter.Result spannerWriteResults =
+
+    CustomTransformation customTransformation =
+        CustomTransformation.builder(
+                options.getTransformationJarPath(), options.getTransformationClassName())
+            .setCustomParameters(options.getTransformationCustomParameters())
+            .build();
+
+    ChangeEventTransformerDoFn changeEventTransformerDoFn =
```

Comment:
we should skip this stage creation unless there is a transformation function supplied

Reply:
This stage is for all sorts of transformation - session file based and custom transformation